File: language-speed-benchmarks/java/Benchmark.java
public class Benchmark {
    public static boolean isPrime(int n) {
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) return false;
        }
        return true;
    }

    public static void main(String[] args) {
        long startNano = System.nanoTime();
        int count = 0;
        for (int i = 2; i < 100000; i++) {
            if (isPrime(i)) count++;
        }
        long endNano = System.nanoTime();
        double elapsedSeconds = (endNano - startNano) / 1e9;
        System.out.printf("Java: %.6f seconds%n", elapsedSeconds);
        System.out.println("Prime count: " + count);
    }
}

File: language-speed-benchmarks/lua/benchmark.lua
function is_prime(n)
    for i = 2, math.sqrt(n) do
        if n % i == 0 then
            return false
        end
    end
    return true
end

local start = os.clock()
local count = 0
for i = 2, 100000 do
    if is_prime(i) then
        count = count + 1
    end
end
local end_time = os.clock()
print("Lua:", end_time - start, "seconds")
